src/python/dbs/apis/dbsClient.py

```diff
@@ -400,7 +411,16 @@ def __callServer(self, method="", params={}, data={}, callmethod='GET', content=
                 print("HTTP={} URL={} method={} params={} data={} headers={}".format(callmethod, self.url, method, params, data, request_headers))
             self.http_response = method_func(self.url, method, params, data, request_headers)
         except HTTPError as http_error:
-            self.__parseForException(http_error)
+            if headerHas(http_error.header, 'Content-Type', 'application/json'):
+                self.__parseForException(http_error)
+            else:
+                msg = "\nURL={}\nCode={}\nMessage={}\nHeader={}\nBody={}\n".format(
+                        http_error.url,
+                        http_error.code,
+                        http_error.msg,
+                        http_error.header,
+                        http_error.body)
+                raise Exception(msg)
```

Comment:
Can you call self.__parseForException instead of raising general Exception ? You may create a json string similar to line 417/msg to pass it to self._parseForException.

Reply:
yes, I can please review again. Now the output looks almost identical but I construct custom HTTPError and pass it to `self.__parseForException`. Here is relevant part of new output:
```
RestClient.ErrorHandling.RestClientExceptions.HTTPError: HTTP Error 400:
URL=https://cmsweb-testbed.cern.ch:8443/dbs/bla/DBSReader/help
Code=400
Message=Bad Request
Header=HTTP/1.1 400 Bad Request
Date: Thu, 11 Nov 2021 18:22:16 GMT
Server: Apache
Content-Length: 226
Connection: close
Content-Type: text/html; charset=iso-8859-1


Body=<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
</p>
</body></html>
```